### PR TITLE
[streams] Add tests for aborting pipeTo()

### PR DESCRIPTION
**Pull request by ricea**
Add tests for aborting ReadableStream.prototype.pipeTo using an
AbortSignal.

Tests correspond to PR whatwg/streams#744.

### CODE REVIEW

File: streams/piping/abort.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+for (const invalidSignal of [null, 'AbortSignal', true, -1, Object.create(AbortSignal.prototype)]) {
+  promise_test(t => {
+    const rs = new ReadableStream({
+      start(controller) {
+        controller.close();
+      }
+    });
+    const ws = new WritableStream();
+    return promise_rejects(t, new TypeError(), rs.pipeTo(ws, { signal: invalidSignal }), 'pipeTo should reject');
```
Comment (domenic) [inline streams/piping/abort.js]: Should we test that this never calls any underlying source methods?

File: streams/piping/abort.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+  abortController.abort();
+  return promise_rejects(t, 'AbortError', rs.pipeTo(ws, { signal }), 'pipeTo should reject')
+      .then(() => Promise.all([
+        rs.getReader().closed,
+        promise_rejects(t, 'AbortError', ws.getWriter().closed, 'writer.closed should reject')
+      ]))
+      .then(() => {
+        assert_equals(rs.events.length, 2, 'cancel should have been called');
+        assert_equals(rs.events[0], 'cancel', 'first event should be cancel');
+        assert_equals(rs.events[1].name, 'AbortError', 'the argument to cancel should be an AbortError');
```
Comment (domenic) [inline streams/piping/abort.js]: Maybe also test that its .constructor.name is DOMException.

**Review by domenic — APPROVED** [review]
Overall LGTM, some minor suggestions. A bit amused that almost all of the tests use an already-aborted signal, but I guess that's where the edge cases lie.

OK, here's a suggestion: wait for the pipe to complete successfully, then abort the signal, and make sure that doesn't somehow error/cancel the streams? I dunno, maybe that's too silly.

File: streams/piping/abort.js (hunk 2 above)
Comment (ricea) [reply streams/piping/abort.js]: Done.

**Comment by ricea** [comment]
I will write tests for abort-after-completion. That was an important oversight.

**Review by domenic — APPROVED** [review]
